src/core/thread/mle.cpp

```diff
@@ -3553,10 +3556,12 @@ void Mle::HandleChildUpdateRequest(RxInfo &aRxInfo)
     }
 #endif
 
-    SuccessOrExit(error = SendChildUpdateResponse(tlvList, challenge));
+    // Send the response to the requester, regardless if it's this device's parent or not
+    aRxInfo.mMessageInfo.GetPeerAddr().GetIid().ConvertToExtAddress(extAddr);
+    SuccessOrExit(error = SendChildUpdateResponse(tlvList, challenge, extAddr));
```

Comment:
We can also consider passing the destination IPv6 address as input to `SendChildUpdateResponse()` ?

I think it will simplify the code as we don;t need to convert it to `ExtAddress` here and then have the `SendChildUpdateResponse()` convert it back to an IPv6 address.

Reply:
Oh yea, I'm seeing that's the style of other functions here. I originally made the change on an older branch that hadn't been updated. No problem, will do!

Reply:
Updated!